### PR TITLE
Replace atrribute with variable and other updates

### PR DESCRIPTION
**Pull request by ilevkivskyi**
In this PR:
- Attributes replaced with variables almost everywhere.
- Formatting changes: internal refs, consistent capitalization in section titles (CapWords for sections, Normal for subsections), and some small changes.
- Moved the paragraph about mutability of `__annotations__` to the section runtime effects.
- Added possible future support for duck-typing (via Protocols as discussed in https://github.com/python/typing/issues/11) as one more argument in favor of the PEP.

Attn: @kirbyfan64 @phouse512 @gvanrossum @lisroach

### CODE REVIEW

File: pep-0526.txt [hunk 1]
```diff
@@ -174,38 +178,23 @@ by a different type::
   a: int
   a: str  # Static type checker will warn about this.
 
-``__annotations__`` is writable, so this is permitted::
+.. _classvar:
```
Comment (gvanrossum) [inline pep-0526.txt:181]: Here it looks like you lost a whole section of new stuff added previously about `__annotations__`.

File: pep-0526.txt [hunk 2]
```diff
@@ -174,38 +178,23 @@ by a different type::
...
+in ``__init__`` or ``__new__``. The proposed syntax is as follows::
 
   class BasicStarship:
       captain: str = 'Picard'               # instance variable with default
       damage: int                           # instance variable without default
       stats: ClassVar[Dict[str, int]] = {}  # class variable
 
 Here ``ClassVar`` is a special class in typing module that indicates to
-static type checker that this attribute should not be set on class instances.
+static type checker that this variable should not be set on class instances.
```
Comment (gvanrossum) [inline pep-0526.txt]: "class instances" -> instances

File: pep-0526.txt [hunk 3]
```diff
@@ -304,22 +297,23 @@ unpacking::
...
 =============================================
 
 - A new covariant type ``ClassVar[T_co]`` is added to the ``typing``
   module. It accepts only a single argument that should be a valid type,
   and is used to annotate class variables that should no be set on class
   instances. This restriction is ensured by static checkers,
-  but not at runtime. See Attribute Annotations for examples
-  and explanations for the usage of ``ClassVar``, and see the Rejected
-  Proposals section for more information on the reasoning behind ``ClassVar``.
+  but not at runtime. See
```
Comment (gvanrossum) [inline pep-0526.txt]: See -> See the

File: pep-0526.txt (hunk 2 above)
Comment (ilevkivskyi) [reply pep-0526.txt]: Done

File: pep-0526.txt (hunk 3 above)
Comment (ilevkivskyi) [reply pep-0526.txt]: Done

File: pep-0526.txt (hunk 1 above)
Comment (ilevkivskyi) [reply pep-0526.txt:181]: I moved this paragraph below to lines 357-371 (in new version) right next to two other paragraphs where `__annotations__` is discussed.

**Comment by ilevkivskyi** [comment]
@gvanrossum Pushed the new version with your comments implemented.

**Pull request merged**